apps/rellotge/rellotge.js

```diff
@@ -0,0 +1,131 @@
+
+    // draw time
+    var time = da[4].substr(0, 5);
+    var hora = time.split(":");
+    if (m >= 0 && m < 2) {
+      t = leshores[d.getHours()] + "\r\nen punt";
+    } else if (m >= 2 && m < 5) {
+      t = leshores[d.getHours()] + "\r\ntocades";
+    } else if (m >= 5 && m < 7) {
+      t = leshores[d.getHours()] + "\r\nben tocades";
+    } else if (m >= 7 && m < 10) {
+      t = "Mig quart\r\n" + leshores2[d.getHours()];
+    } else if (m >= 10 && m < 12) {
+      t = "Mig quart\r\ntocat\r\n" + leshores2[d.getHours()];
+    } else if (m >= 12 && m < 15) {
+      t = "Mig quart\r\nben tocat\r\n" + leshores2[d.getHours()];
+    } else if (m >= 15 && m < 17) {
+      t = "Un quart\r\n" + leshores2[d.getHours()];
+    } else if (m >= 17 && m < 20) {
+      t = "Un quart\r\ntocat\r\n" + leshores2[d.getHours()];
+    } else if (m >= 20 && m < 22) {
+      t = "Un quart\r\nben tocat\r\n" + leshores2[d.getHours()];
+    } else if (m >= 22 && m < 25) {
+      t = "Un quart i mig\r\n" + leshores2[d.getHours()];
+    } else if (m >= 25 && m < 27) {
+      t = "Un quart i mig\r\ntocat\r\n" + leshores2[d.getHours()];
+    } else if (m >= 27 && m < 30) {
+      t = "Un quart i mig\r\nben tocat\r\n" + leshores2[d.getHours()];
+    } else if (m >= 30 && m < 32) {
+      t = "Dos quarts\r\n" + leshores2[d.getHours()];
+    } else if (m >= 32 && m < 35) {
+      t = "Dos quarts\r\ntocats\r\n" + leshores2[d.getHours()];
+    } else if (m >= 35 && m < 37) {
+      t = "Dos quarts\r\nben tocats\r\n" + leshores2[d.getHours()];
+    } else if (m >= 37 && m < 40) {
+      t = "Dos quarts i mig\r\n" + leshores2[d.getHours()];
+    } else if (m >= 40 && m < 42) {
+      t = "Dos quarts i mig\r\ntocats\r\n" + leshores2[d.getHours()];
+    } else if (m >= 42 && m < 45) {
+      t = "Dos quarts i mig\r\nben tocats\r\n" + leshores2[d.getHours()];
+    } else if (m >= 45 && m < 47) {
+      t = "Tres quarts\r\n" + leshores2[d.getHours()];
+    } else if (m >= 47 && m < 50) {
+      t = "Tres quarts\r\ntocats\r\n" + leshores2[d.getHours()];
+    } else if (m >= 50 && m < 52) {
+      t = "Tres quarts\r\nben tocats\r\n" + leshores2[d.getHours()];
+    } else if (m >= 52 && m < 55) {
+      t = "Tres quarts i mig\r\n" + leshores2[d.getHours()];
+    } else if (m >= 55 && m < 57) {
+      t = "Tres quarts i mig\r\ntocats\r\n" + leshores2[d.getHours()];
+    } else if (m >= 57) {
+      t = "Tres quarts i mig\r\nben tocats\r\n" + leshores2[d.getHours()];
+    }
+    g.setFont(font, timeFontSize);
+    g.drawString(t, xyCenter, yposTime, true);
+
+    // draw Day, name of month, Date
+    var date = [da[0], da[1], da[2]].join(" ");
+    g.setFont(font, dateFontSize);
+    
+    var mu = "";
+    if (m < 10) {mu = "0"+m;} else {mu = m;}
+
+    g.drawString(d.getHours()+":"+mu, xyCenter, yposDate, true);
+
+
+  }
+
+  // handle switch display on by pressing BTN1
+  Bangle.on('lcdPower', function(on) {
+    if (on) {
+      Bangle.loadWidgets();
+      Bangle.drawWidgets();
```

Comment:
We only need to do this once, then we can leave the widgets to draw themselves. Perhaps:

```javascript
  function onLcd(on) {
    if (on) {
      Bangle.loadWidgets();
      Bangle.drawWidgets();
      //drawSimpleClock();
      Bangle.removeListener('lcdPower', onLcd);
    }
  }
  Bangle.on('lcdPower', onLcd);
```